core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -1334,15 +1335,28 @@ protected void loadOrPrepareLM(boolean closeEarly) {
      * Internal method to clean up the graph.
      */
     protected void cleanUp() {
-        int prevNodeCount = ghStorage.getNodes();
-        PrepareRoutingSubnetworks preparation = new PrepareRoutingSubnetworks(ghStorage, encodingManager.fetchEdgeEncoders());
+        PrepareRoutingSubnetworks preparation = new PrepareRoutingSubnetworks(ghStorage, buildSubnetworkRemovalJobs());
         preparation.setMinNetworkSize(minNetworkSize);
         preparation.doWork();
-        int currNodeCount = ghStorage.getNodes();
-        logger.info("edges: " + Helper.nf(ghStorage.getEdges()) + ", nodes " + Helper.nf(currNodeCount)
-                + ", there were " + Helper.nf(preparation.getMaxSubnetworks())
-                + " subnetworks. removed them => " + Helper.nf(prevNodeCount - currNodeCount)
-                + " less nodes");
+        logger.info("nodes: " + Helper.nf(ghStorage.getNodes()) + ", edges: " + Helper.nf(ghStorage.getEdges()));
+    }
+
+    private List<PrepareJob> buildSubnetworkRemovalJobs() {
+        List<FlagEncoder> encoders = encodingManager.fetchEdgeEncoders();
+        List<PrepareJob> jobs = new ArrayList<>();
+        for (FlagEncoder encoder : encoders) {
+            // for encoders with turn costs we do an edge-based subnetwork removal, because they *might* be used with
+            // a profile with turn_costs=true
```

Comment:
I made this depend on the encoder, so it might happen that we have multiple profiles (some with and some without turn costs) that use the same encoder (the same access flags) and so we also use the edge-based subnetwork removal for the profiles without turn costs. This means we might remove some subnetworks that are not accessible if we include turn costs also for profiles that ignore turn costs anyway. But this does not seem to be a big problem to me. The (more general) alternative would be running the subnetwork removal *per profile*, but then we also need access flags *per profile*. Or maybe no access flags at all but this is another story so I went with the simple solution here. The other extreme would be *always* using the edge-based subnetwork removal (to make things more simple).